### ISSUE TITLE
Please support ꜱᴡᴄ 127

### ISSUE THREAD

**Issue by ytrezq** [ConsenSys/mythril#1258]
## Description :

Detects the possibility to ᴊᴜᴍᴘ at arbitrary address either because of low level functions used in source code or for well written and audited solidity code but with compiler bugs like `OptimizerStateKnowledgeNotResetForJumpdest` or `OptimizerClearStateOnCodePathJoin` and of course, not simply if they are in the code bug which input sequence to trigger them.

## Note :

A better thing would be to leverage those issues in order to trigger more serious one like being able to withdraw ether or fire a token `Transfer` event without background checks.

## Tests ;

For compiler bugs I couldn’t find the issue number in the GitHub project nor the commit which fixed them so I couldn’t get enough details for a test case.

**Comment by b-mueller** [comment]
@ytrezq #1268 introduces this feature, would you like to test it?

**Comment by ytrezq** [comment]
@b-mueller if it doesn’t detect almost arbitrary jumps from the call graph, then no I can’t. What I’m requesting doesn’t only require adding a module. It also requires adding the capability to Laser accordingly (otherwise it should detects only bugs from solidity code and not low level compiler bugs).

**Comment by norhh** [comment]
`otherwise it should detects only bugs from solidity code and not low level compiler bugs` Hey, mythril is a bytecode analyzer, it doesn't know the difference between solidity source code and bytecode, so there is no need to change anything in laser. Also #1269 is the arbitrary jump feature.

**Comment by ytrezq** [comment]
@norhh yes, but based on a call graph. If the call graph say that a jump is static whereas it could be dynamic then nothing will happen.

**Comment by norhh** [comment]
What I am checking is that if the jump destination is a symbol rather than a concrete value then a multilocation jump is possible.

**Comment by ytrezq** [comment]
We don’t have a minimal example of `OptimizerClearStateOnCodePathJoin` and `OptimizerStateKnowledgeNotResetForJumpdest`. My test case are taking days.

Any ideas of their bug number in https://github.com/ethereum/solidity/issues?

**Comment by ytrezq** [comment]
Ok, here’s the test cases : https://github.com/ethereum/solidity/issues/692 and https://github.com/ethereum/solidity/issues/1416 (the latter being harder to trigger). But I don’t have my personal machines right now (nor in the future) so I can’t compile and run Mythril on those examples.
